AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+/// Defines the contract for an AuthSession that can vend AWS credentials and store a user ID
+/// (`sub`) for the underlying OIDC-compliant authentication provider such as Cognito user pools.
+/// Concrete types that use Cognito identity pools to obtain AWS credentials can also vend the
+/// associated identityID.
+///
+/// **The `isSignedIn` property**
+///
+/// The `AuthSession.isSignedIn` property encapsulates a number of different decisions to provide a high-level
+/// answer to the question "is the current user signed in?" However, "signed in" can mean different things depending on
+/// the authentication provider, the state of the authentication tokens, and even whether the network is reachable. Types
+/// that conform to ``AWSAuthSessionBehavior`` must implement the `isSignedIn` flag to conform to the
+/// following rules:
+/// TODO: Add detailed discussion of isSignedIn for Cognito. Include discussion of cases left deliberately up to the discretion
+/// of the plugin developer.
+public protocol AWSAuthSessionBehavior<OIDCCredentials> : AuthSession {
+
+    /// The concrete type holding the OIDC tokens from the authentication provider. Generally, this type will have at least
+    /// methods for retrieving an identity token and an access token.
+    associatedtype OIDCCredentials
```

Comment:
how about we rename this to `TokensType`

Reply:
Counteroffer: `Tokens` ?

Reply:
Sounds good